src/main/java/com/openlattice/authorization/EdmAuthorizationHelper.java

```diff
@@ -330,12 +329,52 @@ public EdmAuthorizationHelper( EdmManager edm, AuthorizationManager authz, Entit
             Set<UUID> entitySetIds,
             EnumSet<Permission> requiredPermissions,
             Set<Principal> principals ) {
-        return ( entitySetIds.isEmpty() )
-                ? Maps.newHashMap()
-                : entitySetIds.stream().collect( Collectors.toMap(
-                Function.identity(),
-                entitySetId -> getAuthorizedPropertyTypes( entitySetId, requiredPermissions, principals )
-        ) );
+        if ( entitySetIds.isEmpty() ) {
+            return Maps.newHashMap();
+        }
+
+        Map<UUID, Map<UUID, PropertyType>> entitySetIdsToPropertyTypes = entitySetManager
+                .getPropertyTypesOfEntitySets( entitySetIds );
+
+        Set<AccessCheck> accessChecks = Sets.newHashSetWithExpectedSize( entitySetIdsToPropertyTypes.size() );
+
+        entitySetIdsToPropertyTypes.entrySet().forEach( entry -> {
+            UUID entitySetId = entry.getKey();
+
+            entry.getValue().keySet().forEach( propertyTypeId -> {
+                accessChecks.add( new AccessCheck( new AclKey( entitySetId, propertyTypeId ), requiredPermissions ) );
+            } );
+        } );
+
+        Map<UUID, Map<UUID, PropertyType>> authorizedEntitySetsToPropertyTypes = Maps
+                .newLinkedHashMapWithExpectedSize( entitySetIdsToPropertyTypes.size() );
+        authz.accessChecksForPrincipals( accessChecks, principals ).forEach( authorization -> {
```

Comment:
Consider using authorize since you don't have to worry about merging repeat access checks and it avoids some unnecessary wrapping of objects.